File: app/src/main/java/com/fastaccess/provider/timeline/handler/EmojiHandler.java
package com.fastaccess.provider.timeline.handler;

import android.text.SpannableStringBuilder;

import com.fastaccess.helper.Logger;
import com.fastaccess.provider.emoji.Emoji;
import com.fastaccess.provider.emoji.EmojiManager;

import net.nightwhistler.htmlspanner.TagNodeHandler;

import org.htmlcleaner.TagNode;

/**
 * Created by Kosh on 27 May 2017, 4:54 PM
 */

public class EmojiHandler extends TagNodeHandler {

    @Override public void handleTagNode(TagNode node, SpannableStringBuilder builder, int start, int end) {
        String emoji = node.getAttributeByName("alias");
        if (emoji != null) {
            Emoji unicode = EmojiManager.getForAlias(emoji);
            if (unicode != null && unicode.getUnicode() != null) {
                builder.replace(start, end, " " + unicode.getUnicode() + " ");
            }
        } else if (node.getText() != null) {
            Logger.e(node.getText());
            Emoji unicode = EmojiManager.getForAlias(node.getText().toString());
            if (unicode != null && unicode.getUnicode() != null) {
                builder.replace(start, end, " " + unicode.getUnicode() + " ");
            }
        }
    }

    @Override public void beforeChildren(TagNode node, SpannableStringBuilder builder) {
        super.beforeChildren(node, builder);
    }
}

File: markdown/src/main/java/com/fastaccess/markdown/spans/EmojiHandler.kt
package com.fastaccess.markdown.spans

import android.text.SpannableStringBuilder
import com.fastaccess.markdown.emoji.EmojiManager
import net.nightwhistler.htmlspanner.SpanStack
import net.nightwhistler.htmlspanner.TagNodeHandler
import org.htmlcleaner.TagNode

/**
 * Created by Kosh on 27 May 2017, 4:54 PM
 */

class EmojiHandler : TagNodeHandler() {
    override fun handleTagNode(node: TagNode?, builder: SpannableStringBuilder?, start: Int, end: Int, spanStack: SpanStack?) {
        val emoji = node?.getAttributeByName("alias")
        if (emoji != null) {
            val unicode = EmojiManager.getForAlias(emoji)
            if (unicode?.unicode != null) {
                builder?.replace(start, end, " " + unicode.unicode + " ")
            }
        } else if (node?.text != null) {
            val unicode = EmojiManager.getForAlias(node.text.toString())
            if (unicode?.unicode != null) {
                builder?.replace(start, end, " " + unicode.unicode + " ")
            }
        }
    }
}
